ring-core/test/ring/middleware/multipart_params/test/multipart_params.clj

```diff
@@ -0,0 +1,27 @@
+(ns ring.middleware.multipart-params.test.multipart-params
+  (:require [clojure.test :refer :all]
+            [ring.middleware.multipart-params :refer :all]
+            [ring.util.response :as response])
+  (:import org.apache.commons.fileupload.InvalidFileNameException))
+
+(deftest wrap-multipart-params-error-test
+  (let [invalid-file-name "foo\\0.bar"
+        error-msg (str "Invalid file name: " invalid-file-name)]
+    (with-redefs [multipart-params-request
```

Comment:
Don't use `with-redefs` here. It would be better to pass in a request with an invalid filename.